single_sync_v3.sh

```diff
@@ -0,0 +1,137 @@
+        if [ "${CONTEXT}" = "null" ]; then
+          if [ ${DRY} = true ]; then
+              echo "    - DRY MODE is active, skipping layers check"
+          else
+              LOCAL_LAYERS=$(docker run --rm quay.io/skopeo/stable:v1.13 inspect -n --override-os linux --override-arch amd64 docker://${SRC}:${LOCAL_TAG} 2> /dev/null | yq .Layers)
+              TARGET_LAYERS=$(docker run --rm quay.io/skopeo/stable:v1.13 inspect -n --override-os linux --override-arch amd64 docker://$(yq e '.images['"${c}"'].destinations[0]' $1):${LOCAL_TAG} 2> /dev/null | yq .Layers)
+
+              diff <(echo ${LOCAL_LAYERS}) <(echo ${TARGET_LAYERS}) > /dev/null
+              AMD64_DIFF=$?
+
+              echo "    - AMD64 diff exit code is: $AMD64_DIFF"
+              
+              # if not multiarch, the value can be the same
+              ARM64_DIFF=$AMD64_DIFF
+              if [ ${MULTI_ARCH} = true ]; then
+                LOCAL_LAYERS=$(docker run --rm quay.io/skopeo/stable:v1.13 inspect -n --override-os linux --override-arch arm64 docker://${SRC}:${LOCAL_TAG} 2> /dev/null | yq .Layers)
+                TARGET_LAYERS=$(docker run --rm quay.io/skopeo/stable:v1.13 inspect -n --override-os linux --override-arch arm64 docker://$(yq e '.images['"${c}"'].destinations[0]' $1):${LOCAL_TAG} 2> /dev/null | yq .Layers)
+
+                diff <(echo ${LOCAL_LAYERS}) <(echo ${TARGET_LAYERS}) > /dev/null
+                ARM64_DIFF=$?
+
+                echo "    - ARM64 diff exit code is: $ARM64_DIFF"
+              fi
+          fi
+        fi
+
+        if [ $AMD64_DIFF -eq 0 ] && [ $ARM64_DIFF -eq 0 ] && [ ${DRY} = false ] && [ ${CONTEXT} = "null" ]; then
+          echo "    - Skipping ${SRC}:${LOCAL_TAG} as it is already synced"
+        else
+          if [ ${DRY} = true ] && [ ${CONTEXT} = "null" ]; then
+            echo "    - DRY MODE is active, skipping pull for ${SRC}:${LOCAL_TAG}"
+          elif [ ${DRY} = true ] && [ ${CONTEXT} != "null" ]; then
+            echo "    - DRY MODE is active, skipping build for ${SRC}:${LOCAL_TAG}"
+          elif [ ${CONTEXT} = "null" ]; then
+            echo "    - Layers are different, syncing ${SRC}:${LOCAL_TAG}"
+            if [ ${MULTI_ARCH} = true ]; then
+              echo "    - No prepull, sync using skopeo"
+            else
+              docker pull ${SRC}:${LOCAL_TAG}
+            fi
+            # 
+          else
+            BUILD_ARGS=""
+            ARG=$(yq e '.images['"${c}"'].build.args | length' $1)
+            for (( a=0; a<${ARG}; a++ ))
+              do
+                ARG_NAME=$(yq e '.images['"${c}"'].build.args['"${a}"'].name' $1)
+                ARG_VALUE=$(yq e '.images['"${c}"'].build.args['"${a}"'].value' $1)
+                BUILD_ARGS=$BUILD_ARGS" --build-arg "$ARG_NAME"="$ARG_VALUE
+            done
+            if [ ${MULTI_ARCH} = true ]; then
+                #--push \
+              TARGET_IMAGE=$(yq e '.images['"${c}"'].destinations[0]' $1):${LOCAL_TAG}
```

Comment:
why is the target image tag not the same as the build when multi-arch is false? i.e. why is not `TARGET_IMAGE=${SRC}:${LOCAL_TAG}`?

Reply:
Because the image should be exactly the one that will be pushed from docker after the build. So it should be something like `registry.sighup.io/fury/PROJECT/NAME:TAG`